audb/core/load.py

```diff
@@ -535,15 +535,27 @@ def _media(
 ) -> typing.Sequence[str]:
 
     if media is None:
-        media = db.files
-    elif isinstance(media, str):
+        return db.files
+    elif len(media) == 0:
+        return []
+
+    if isinstance(media, str):
         pattern = re.compile(media)
-        media = []
+        requested_media = []
         for m in db.files:
             if pattern.search(m):
-                media.append(m)
+                requested_media.append(m)
+        if len(requested_media) == 0:
+            raise ValueError(f"Could not find a media file matching '{media}'")
+    else:
+        requested_media = media
+        for media in requested_media:
```

Comment:
Maybe use `m` instead of `media` as in the `if` case to not overwrite `media'?

Reply:
I changed to `media_file` as we use inside `load_media()`